src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,6 +36,55 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
         this.value = address;
+        block=new Block(this.segmentAddress(address, BLOCK));
+        unit=new Unit(this.segmentAddress(address, UNIT));
+        street=new Street(this.segmentAddress(address, STREET));
+        postalCode=new PostalCode(this.segmentAddress(address, POSTAL_CODE));
+    }
+    
+    
+    
+    /**
+     * Segments and returns either BLOCK, STREET, UNIT, or POSTAL_CODE  
+     */
+    
+    public String segmentAddress(String address, int segment){
+    	int commaCount=0;
+    	int beginIndex=2;
+    	int endIndex;
+    	int i;
+    	String target="";
+    	
+    	for(i=2;i<address.length();i++){
+    		if(address.charAt(i)==','){
+    			commaCount++;
+    			if(commaCount==segment){
+    				endIndex=i;
+    				target= address.substring(beginIndex, endIndex);
+    				return target;
+    			}
+    			beginIndex=i+2;
+    		}	
+    	}
+    	endIndex=i;
+		target=address.substring(beginIndex, endIndex);
+		return target;
+    }
```

Comment:
notice how this is harder to understand (can consider better naming for variables and clearer comments- what are BLOCK, STREET etc.- the constant int?) than the example we went through in tutorial. also, have you considered what happens when fields are missing?

remember to update the validation method, tests and documentation as well!